src/meepgeom.cpp

```diff
@@ -1747,8 +1750,18 @@ void fragment_stats::update_stats_from_material(material_type mat, size_t pixels
       count_nonzero_conductivity_pixels(med, pixels);
       break;
     }
+    case material_data::MATERIAL_USER: {
+      for (int i = 0; i < extra_materials.num_items; ++i) {
+        medium_struct *med = &extra_materials.items[i]->medium;
+        if (!anisotropic_pixels_already_added) { count_anisotropic_pixels(med, pixels); }
+        count_nonlinear_pixels(med, pixels);
+        count_susceptibility_pixels(med, pixels);
+        count_nonzero_conductivity_pixels(med, pixels);
+        break;
```

Comment:
This doesn't seem right: it overcounts the pixels if you have e.g. multiple nonlinear or anisotropic materials in the extra_materials list.

Probably set flags for each kind of special medium and only count the first one that is found.

Unfortunately, if we have multiple susceptibility types in the extra_materials we don't know how to count them, e.g. which one to use, but we might as well just use the first one for now.

Reply:
You can modify e.g. `count_nonlinear_pixels` to return a `bool` saying whether anything was counted, so that you don't have to duplicate the check for nonlinearity.